### PR TITLE
[video_player] Updates minimum supported SDK

### PR DESCRIPTION
**Pull request by abdelaziz-mahdy**
fixes https://github.com/flutter/flutter/issues/154050 and https://github.com/flutter/flutter/issues/154054 by setting the min sdk version 
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by matanlurey** [comment]
Thanks. This won't work as you'd expect due to how versioning is resolved, but now that https://github.com/flutter/packages/pull/7497 is merged (the revert) if you send me a roll-forward with the SDK constraints set properly I'd love to LGTM it 😁

**Comment by abdelaziz-mahdy** [comment]
is this the correct way?

**Review by matanlurey — APPROVED** [review]

**Comment by matanlurey** [comment]
> is this the correct way?

Looks good except as per the failing check the minimum Dart bounds needs to be bumped as well, then we can merge!

**Comment by abdelaziz-mahdy** [comment]
> > is this the correct way?
> 
> Looks good except as per the failing check the minimum Dart bounds needs to be bumped as well, then we can merge!

done and updated changelog to reflect it too

**Review by nate-thegrate — APPROVED** [review]

**Comment by stuartmorgan** [comment]
Reviewers: In the future, please include the PR title and description in your review. As someone who uses `git log` and `git blame` frequently in this repository, having PRs landed with extremely misleading descriptions makes life much more difficult. The primary effect of this PR is not to change the minimum SDK, it is to change the rendering pipeline.

File: packages/video_player/video_player_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,15 @@
+## 2.7.2
+
+* Updates minimum supported SDK version to Flutter 3.24/Dart 3.5.
+
+* Re-adds Impeller support.
```
[the remark this answers is not in the document]
Comment (abdelaziz-mahdy) [reply packages/video_player/video_player_android/CHANGELOG.md:5]: Oh I am sorry for that want me to issue a pr fixing that?

**Comment by abdelaziz-mahdy** [comment]
> Reviewers: In the future, please include the PR title and description in your review. As someone who uses `git log` and `git blame` frequently in this repository, having PRs landed with extremely misleading descriptions makes life much more difficult. The primary effect of this PR is not to change the minimum SDK, it is to change the rendering pipeline.

Well the pr started as updating the minimum sdk, so I am sorry for forgetting to update the title to the new changes

**Comment by nate-thegrate** [comment]
@stuartmorgan thanks for letting us know.

In the future, I'll be a bit more judicious about submitting a "rubber-stamp LGTM" 😄

File: packages/video_player/video_player_android/CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/video_player/video_player_android/CHANGELOG.md:5]: No, CHANGELOG formatting isn't something that we would do another release for unless it were extremely broken.

[the remark this answers is not in the document]
Comment (abdelaziz-mahdy) [reply packages/video_player/video_player_android/CHANGELOG.md:5]: Ok thank you for letting me know I will make sure it's correct next time.

**Comment by stuartmorgan** [comment]
> Well the pr started as updating the minimum sdk

I understand that, but in 6 months when someone is trying to understand the origin of this code by looking at `blame` or `log`, what matters is what the PR as landed does. The PR title and description become part of the permanent repository history and are frequently used to understand changes, so should be reviewed just like other documentation in a PR (e.g., code comments).

**Comment by abdelaziz-mahdy** [comment]
> > Well the pr started as updating the minimum sdk
> 
> I understand that, but in 6 months when someone is trying to understand the origin of this code by looking at `blame` or `log`, what matters is what the PR as landed does. The PR title and description become part of the permanent repository history and are frequently used to understand changes, so should be reviewed just like other documentation in a PR (e.g., code comments).

I understand now, and will keep it in mind next time I open a pr to make sure it's correct,

Thank you for explaining the problem

**Pull request merged**